protocols/raft/src/test/java/io/atomix/protocols/raft/protocol/TestRaftProtocol.java

```diff
@@ -16,20 +16,39 @@
 package io.atomix.protocols.raft.protocol;
 
 import io.atomix.cluster.NodeId;
+import io.atomix.utils.concurrent.Scheduled;
+import io.atomix.utils.concurrent.ThreadContext;
 
+import java.time.Duration;
 import java.util.Collection;
 import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeoutException;
 
 /**
  * Base class for Raft protocol.
  */
 public abstract class TestRaftProtocol {
   private final Map<NodeId, TestRaftServerProtocol> servers;
   private final Map<NodeId, TestRaftClientProtocol> clients;
+  private final ThreadContext context;
 
-  public TestRaftProtocol(Map<NodeId, TestRaftServerProtocol> servers, Map<NodeId, TestRaftClientProtocol> clients) {
+  public TestRaftProtocol(
+      Map<NodeId, TestRaftServerProtocol> servers,
+      Map<NodeId, TestRaftClientProtocol> clients,
+      ThreadContext context) {
     this.servers = servers;
     this.clients = clients;
+    this.context = context;
+  }
+
+  <T> CompletableFuture<T> scheduleTimeout(CompletableFuture<T> future) {
+    Scheduled scheduled = context.schedule(Duration.ofMillis(100), () -> {
```

Comment:
No response under 100 ms results in timeout?

Reply:
They're all local in tests anyways, so I suspect a 100ms response time from another thread would be indicative of a problem anyways.

Reply:
Or an overloaded Travis node ;)

Reply:
touché